tests/notifier/notifier_test.py

```diff
@@ -120,5 +120,140 @@ def test_notifications_for_nonempty_violations(
+        self.assertEquals(
+            'No "inventory" configuration for notifier.',
+            mock_logger.info.call_args[0][0])
+
+    @mock.patch('google.cloud.forseti.notifier.notifier.LOGGER', autospec=True)
+    def test_no_inventory_summary_in_config(self, mock_logger):
+        mock_service_config = mock.MagicMock()
+        mock_service_config.get_notifier_config.return_value = dict(
+            inventory=dict(blah=1))
+        notifier.run_inv_summary('blah', mock_service_config)
+        self.assertTrue(mock_logger.info.called)
+        self.assertEquals(
+            'No "inventory summary" configuration for notifier.',
+            mock_logger.info.call_args[0][0])
+
+    @mock.patch('google.cloud.forseti.notifier.notifier.LOGGER', autospec=True)
+    def test_inventory_summary_not_enabled_in_config(self, mock_logger):
+        mock_service_config = mock.MagicMock()
+        mock_service_config.get_notifier_config.return_value = dict(
+            inventory=dict(summary=dict(enabled=False)))
+        notifier.run_inv_summary('blah', mock_service_config)
+        self.assertTrue(mock_logger.info.called)
+        self.assertEquals(
+            'Inventory summary notifications are turned off.',
+            mock_logger.info.call_args[0][0])
+
+    @mock.patch('google.cloud.forseti.notifier.notifier.LOGGER', autospec=True)
+    def test_inventory_summary_gcs_path_not_set_in_config(self, mock_logger):
+        mock_service_config = mock.MagicMock()
+        mock_service_config.get_notifier_config.return_value = dict(
+            inventory=dict(summary=dict(enabled=True)))
+        notifier.run_inv_summary('blah', mock_service_config)
+        self.assertTrue(mock_logger.error.called)
+        self.assertEquals(
+            '"gcs_path" not set for inventory summary notifier.',
+            mock_logger.error.call_args[0][0])
+
+    @mock.patch('google.cloud.forseti.notifier.notifier.LOGGER', autospec=True)
+    def test_inventory_summary_no_summary_data(self, mock_logger):
+        mock_inv_index = mock.MagicMock()
+        mock_inv_index.notified_at_datetime = None
+        mock_inv_index.get_summary.return_value = dict()
+
+        mock_session = mock.MagicMock()
+        mock_session.query.return_value.get.return_value = mock_inv_index
+
+        mock_service_config = mock.MagicMock()
+        mock_service_config.scoped_session.return_value.__enter__.return_value = mock_session
+        mock_service_config.get_notifier_config.return_value = dict(
+            inventory=dict(summary=dict(enabled=True, gcs_path='gs://xx')))
+
+        notifier.run_inv_summary('blah', mock_service_config)
+        self.assertTrue(mock_logger.warn.called)
+        self.assertEquals(
+            'No inventory summary data found.',
+            mock_logger.warn.call_args[0][0])
+
+    @mock.patch(
+        'google.cloud.forseti.notifier.notifier.InventorySummary', autospec=True)
+    def test_inv_summary_data_passed(self, mock_notifier):
```

Comment:
A better name `test_inv_summary_can_run_successfully()` as that is what you are testing here, which is not just about the data.